cmd/util/ledger/migrations/cadence_value_diff.go

```diff
@@ -0,0 +1,745 @@
+	if err != nil {
+		return fmt.Errorf("failed to create runtime with new state payloads: %w", err)
+	}
+
+	// Iterate through all domains and compare cadence values.
+	for _, domain := range domains {
+		err := dr.diffStorageDomain(oldRuntime, newRuntime, domain)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (dr *CadenceValueDiffReporter) diffStorageDomain(oldRuntime, newRuntime *readonlyStorageRuntime, domain string) error {
+
+	oldStorageMap := oldRuntime.Storage.GetStorageMap(dr.address, domain, false)
+
+	newStorageMap := newRuntime.Storage.GetStorageMap(dr.address, domain, false)
+
+	if oldStorageMap == nil && newStorageMap == nil {
+		// No storage maps for this domain.
+		return nil
+	}
+
+	if oldStorageMap == nil && newStorageMap != nil {
+		dr.reportWriter.Write(
+			difference{
+				Address: dr.address.Hex(),
+				Domain:  domain,
+				Kind:    diffKindString[storageMapExistDiffKind],
+				Msg: fmt.Sprintf(
+					"old storage map doesn't exist, new storage map has %d elements with keys %v",
+					newStorageMap.Count(),
+					getStorageMapKeys(newStorageMap),
+				),
+			})
+
+		return nil
+	}
+
+	if oldStorageMap != nil && newStorageMap == nil {
+		dr.reportWriter.Write(
+			difference{
+				Address: dr.address.Hex(),
+				Domain:  domain,
+				Kind:    diffKindString[storageMapExistDiffKind],
+				Msg: fmt.Sprintf(
+					"new storage map doesn't exist, old storage map has %d elements with keys %v",
+					oldStorageMap.Count(),
+					getStorageMapKeys(oldStorageMap),
+				),
+			})
+
+		return nil
+	}
+
+	oldKeys := getStorageMapKeys(oldStorageMap)
+	newKeys := getStorageMapKeys(newStorageMap)
+
+	onlyOldKeys, onlyNewKeys, sharedKeys := diff(oldKeys, newKeys)
+
+	// Log keys only present in old storage map
+	if len(onlyOldKeys) > 0 {
+		dr.reportWriter.Write(
+			difference{
+				Address: dr.address.Hex(),
+				Domain:  domain,
+				Kind:    diffKindString[storageMapKeyDiffKind],
+				Msg: fmt.Sprintf(
+					"old storage map has %d elements with keys %v, that are not present in new storge map",
+					len(onlyOldKeys),
+					onlyOldKeys,
+				),
+			})
+	}
+
+	// Log keys only present in new storage map
+	if len(onlyNewKeys) > 0 {
+		dr.reportWriter.Write(
+			difference{
+				Address: dr.address.Hex(),
+				Domain:  domain,
+				Kind:    diffKindString[storageMapKeyDiffKind],
+				Msg: fmt.Sprintf(
+					"new storage map has %d elements with keys %v, that are not present in old storge map",
+					len(onlyNewKeys),
+					onlyNewKeys,
+				),
+			})
+	}
+
+	// Compare elements present in both storage maps
+	for _, key := range sharedKeys {
+
+		trace := fmt.Sprintf("%s[%v]", domain, key)
+
+		var mapKey interpreter.StorageMapKey
+
+		switch key := key.(type) {
+		case interpreter.StringAtreeValue:
+			mapKey = interpreter.StringStorageMapKey(key)
+
+		case interpreter.StringStorageMapKey:
+			mapKey = key
+
+		case interpreter.Uint64StorageMapKey:
+			mapKey = key
```

Comment:
Do we actually run into these cases? 

I guess we also need a case for `interpreter.Uint64AtreeValue`

Reply:
> Do we actually run into these cases?

I remember running into `interpreter.StringAtreeValue` during testing.

> I guess we also need a case for interpreter.Uint64AtreeValue

Good catch! Done in [d977b35](https://github.com/onflow/flow-go/pull/5499/commits/d977b352821a2180db8f1f9c0c41c484c9b56fbc)